### PR TITLE
Environment Setup

### PR DESCRIPTION
**Pull request by mendesfabio**
This PR sets up our development environment. I changed the package manager to Yarn (because it's faster than NPM), added some style guides resources, and created a folder structure pattern.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,9 @@
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+indent_size = 2
+indent_style = space
+insert_final_newline = true
+trim_trailing_whitespace = true
```
Comment (ribeirojose) [inline .editorconfig:9]: 😢

File: gatsby-config.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module.exports = {
   siteMetadata: {
-    title: `Gatsby Default Starter`,
+    title: `NEO Empresarial`,
     description: `Kick off your next, great Gatsby project with this default starter. This barebones starter ships with the main Gatsby configuration files you might need.`,
```
Comment (ribeirojose) [inline gatsby-config.js:4]: Pretty sure description and author are not those as well :P

File: package.json [hunk 3]
```diff
@@ -5,18 +5,21 @@
   "version": "0.1.0",
   "author": "Kyle Mathews <mathews.kyle@gmail.com>",
```
Comment (ribeirojose) [inline package.json:6]: Is this the author for this package?

File: .editorconfig (hunk 1 above)
Comment (mendesfabio) [reply .editorconfig]: I don't know what's going on AHAHAHAHA I've just committed a newline at the end of the file but is not here.

File: package.json (hunk 3 above)
Comment (mendesfabio) [reply package.json:6]: I think he developed the starter template that @leticiapamplona used to create our project.

**Comment by mendesfabio** [comment]
> I have outlined some small things that would be nice to haves. Is there any reason you opted for JS instead of TS?

I don't have hahaha But, other members didn't work before with either Web Development or JavaScript itself. I don't know whether it would be a good idea to start with TS.

**Comment by ribeirojose** [comment]
I think it'd be a good idea to start with TS, and if there're any issues, you can always turn off some parts of it in .tsconfig. But I'll leave that to you =)

**Pull request merged**